src/modules/aws_codepipeline/mappers/pipeline_declaration.ts

```diff
@@ -107,19 +106,34 @@ export class PipelineDeclarationMapper extends MapperBase<PipelineDeclaration> {
         name,
       },
       async (cl, cmd) => {
-        let allSuccess = true;
+        let pipelinePending = true;
         try {
           const data = await cl.getPipelineState(cmd);
+          let succeededStates = 0;
           if (data.stageStates && data.stageStates.length > 0) {
             for (const state of data.stageStates) {
-              const latest = state.latestExecution;
-              if (latest?.status !== 'Succeeded') {
-                allSuccess = false;
-                break;
+              // first we check if there is any failure
+              if (state && state.actionStates) {
+                for (const action of state.actionStates) {
+                  const latestStatus = action.latestExecution?.status;
+                  if (
+                    latestStatus &&
+                    ['Cancelled', 'Stopped', 'Superseeded', 'Failed'].includes(latestStatus)
+                  ) {
+                    // pipeline has failed, we can stop it
+                    pipelinePending = false;
+                    break;
```

Comment:
this will break the inner loop on `state.actionStates`, but if `pipelinePending = false` do we want to also `break` the `data.stageStates` loop?